internal/services/gitserver/main.go

```diff
@@ -139,8 +140,11 @@ func NewGitserver(ctx context.Context, l *zap.Logger, c *config.Gitserver) (*Git
 	}
 	log = logger.Sugar()
 
+	repoCleaner := NewRepoCleaner(c, log)
```

Comment:
A repo lock used by the repo cleaner and the various handlers is probably needed or it'll cause some concurrency issues if repo cleaner runs when someone at the same time is pushing.

Probably this is a very remote probability some for the moment you can just add a TODO and remember to open an issue when this PR will be merged).

Reply:
Please add a comment to remember that we should Add a lock